### PR TITLE
Add Array * Scalar

### PR DESCRIPTION
**Pull request by oxinabox**
Nabla has this, we should have it too.

I am sure it can be extended to work for complex but i don't know how so am happy to make it a follow up PR

### CODE REVIEW

File: src/rulesets/LinearAlgebra/dense.jl [hunk 1]
```diff
@@ -149,6 +149,21 @@ function rrule(::typeof(*), A::AbstractMatrix{<:Real}, B::AbstractMatrix{<:Real}
     return A * B, times_pullback
 end
 
+function rrule(::typeof(*), A::Real, B::AbstractArray{<:Real})
+    function times_pullback(Ȳ)
+        return (NO_FIELDS, @thunk(dot(Ȳ, B)), @thunk(A' * Ȳ))
```
Comment (mzgubic) [inline src/rulesets/LinearAlgebra/dense.jl:154]: Why `A'` rather than just `A`?

File: test/rulesets/LinearAlgebra/dense.jl [hunk 2]
```diff
@@ -115,13 +115,21 @@
...
     end
     @testset "*" begin
-        dims = [3,4,5]
-        for n in dims, m in dims, p in dims
-            n > 3 && n == m == p && continue  # don't need to test square case multiple times
-            A = randn(m, n)
-            B = randn(n, p)
-            Ȳ = randn(m, p)
-            rrule_test(*, Ȳ, (A, randn(m, n)), (B, randn(n, p)))
+        @testset "Matrix-Matrix" begin
```
Comment (mzgubic) [inline test/rulesets/LinearAlgebra/dense.jl:118]: Out of scope perhaps but are there tests for Vector*Matrix anywhere? `dot` is tested but I don't think `*` is

**Review by mzgubic — APPROVED** [review]
LGTM

File: src/rulesets/LinearAlgebra/dense.jl (hunk 1 above)
Comment (oxinabox) [reply src/rulesets/LinearAlgebra/dense.jl:154]: Yeah fair, with this restricted to `Real` it is the same.

```suggestion
        return (NO_FIELDS, @thunk(dot(Ȳ, B)), @thunk(A * Ȳ))
```

File: test/rulesets/LinearAlgebra/dense.jl (hunk 2 above)
Comment (oxinabox) [reply test/rulesets/LinearAlgebra/dense.jl:118]: I am not sure we actually have a matric vector multiply `rrule` defined.

File: src/rulesets/LinearAlgebra/dense.jl [hunk 3]
```diff
@@ -149,6 +149,21 @@ function rrule(::typeof(*), A::AbstractMatrix{<:Real}, B::AbstractMatrix{<:Real}
...
+    end
+    return A * B, times_pullback
+end
+
+function rrule(::typeof(*), B::AbstractArray{<:Real}, A::Real)
+    function times_pullback(Ȳ)
+        return (NO_FIELDS, @thunk(A' * Ȳ), @thunk(dot(Ȳ, B)))
```
Comment (oxinabox) [inline src/rulesets/LinearAlgebra/dense.jl:161]:
```suggestion
        return (NO_FIELDS, @thunk(A * Ȳ), @thunk(dot(Ȳ, B)))
```

File: test/rulesets/LinearAlgebra/dense.jl (hunk 2 above)
Comment (mzgubic) [reply test/rulesets/LinearAlgebra/dense.jl:118]: cool I would be keen to tackle this

**Pull request merged**